contracts/deposit/ExecuteDepositUtils.sol

```diff
@@ -71,18 +71,23 @@ library ExecuteDepositUtils {
         bool balanceWasImproved;
         uint256 marketTokensSupply;
         EventUtils.EventLogData callbackEventData;
+        MarketPoolValueInfo.Props poolValueInfo;
     }
 
     address public constant RECEIVER_FOR_FIRST_DEPOSIT = address(1);
 
     // @dev executes a deposit
     // @param params ExecuteDepositParams
-    function executeDeposit(IExecuteDepositUtils.ExecuteDepositParams memory params, Deposit.Props memory deposit) external returns (uint256 receivedMarketTokens) {
+    // @param deposit
+    // @param skipRemoval if true, the deposit will not be removed from the data store.
+    // This is used when executing a deposit as part of a shift or a glv deposit and the deposit is not stored in the data store
+    function executeDeposit(IExecuteDepositUtils.ExecuteDepositParams memory params, Deposit.Props memory deposit, bool skipRemoval) external returns (uint256 receivedMarketTokens) {
         // 63/64 gas is forwarded to external calls, reduce the startingGas to account for this
         params.startingGas -= gasleft() / 63;
 
-        DepositStoreUtils.remove(params.dataStore, params.key, deposit.account());
-
+        if (!skipRemoval) {
+            DepositStoreUtils.remove(params.dataStore, params.key, deposit.account());
```

Comment:
wondering if there should be a validation here, e.g.
```
if (dataStore.containsBytes32(Keys.SHIFT_LIST, key)) {
  revert Errors.RemovalShouldNotBeSkipped(Keys.SHIFT_LIST, key);
}
```

just to make it more difficult to incorrectly skip removal when code is updated